pkg/awsutils/awsutils.go

```diff
@@ -734,9 +734,19 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 		v6cidr, err := cache.imds.GetSubnetIPv6CIDRBlocks(ctx, eniMAC)
 		if err != nil {
 			awsAPIErrInc("GetSubnetIPv6CIDRBlocks", err)
-			return ENIMetadata{}, err
+			// Check if this is a 404 error (e.g., cross-VPC IPv4-only subnet)
+			// If so, handle it gracefully by continuing with empty IPv6 CIDR
+			if isNotFoundError(err) {
+				log.Debugf("IPv6 CIDR not found for ENI %s (likely IPv4-only subnet), continuing with empty IPv6 CIDR", eniMAC)
+				// subnetV6Cidr remains empty, continue processing
+			} else {
+				return ENIMetadata{}, err
+			}
 		} else {
-			subnetV6Cidr = v6cidr.String()
+			// Handle the case where GetSubnetIPv6CIDRBlocks returns empty IPNet for IPv4-only subnets
+			if v6cidr.IP != nil {
```

Comment:
You aren't checking nil checks when you return nil, nil in case the imds data doesn't exist